### PR TITLE
Change scrape_duration_seconds to gauge

### PR DESCRIPTION
**Pull request by carlpett**
Currently the `scrape_duration_seconds` metric is a summary. When trying to reason about #89, it is hard to look back in time to see if others have this problem as well, since we do not have the buckets. I see no real benefit in using a summary here, and think we should be using a histogram instead.

Probably we should be setting the `Buckets` option as well, rather than using the defaults. My lowest 0.5 quantile with the current summary is ~0.04s, so I would make an initial guess that something like `[0.01, 0.015, 0.03, 0.1, 0.15, 0.3, 1.0, 3.0, Inf]` would capture the interesting ranges with acceptable precision?

### CODE REVIEW

**Comment by martinlindhe** [comment]
So, the bucket gets configured by the user and not by us?

Reading up on https://prometheus.io/docs/practices/histograms/

**Comment by brian-brazil** [comment]
This should be a gauge, which contains the information for that scrape.

This was fixed a while back on the node exporter, I'm guessing this was copied from before then.

**Comment by carlpett** [comment]
@brian-brazil Ah, that makes sense. Yes, the summary is probably from the initial commit, more or less. 
I changed to a gauge now, as well as split the result into a separate gauge.

File: exporter.go [hunk 1]
```diff
@@ -81,16 +92,15 @@ func execute(name string, c collector.Collector, ch chan<- prometheus.Metric) {
 	begin := time.Now()
 	err := c.Collect(ch)
 	duration := time.Since(begin)
-	var result string
 
 	if err != nil {
 		log.Errorf("ERROR: %s collector failed after %fs: %s", name, duration.Seconds(), err)
-		result = "error"
+		scrapeSuccess.WithLabelValues(name).Set(0)
```
Comment (brian-brazil) [inline exporter.go]: This should use ConstMetrics, so there's no interference from any scrapes happening at the same time

Comment (carlpett) [inline exporter.go]: Thanks. Fixed!

**Comment by brian-brazil** [comment]
:+1:

**Comment by martinlindhe** [comment]
LGTM!

**Pull request merged**